tests/unit/adaptive_graph_of_thoughts/domain/models/test_graph_elements.py

```
@@ -200,427 +214,289 @@ def check_equality():
-            
-        elem1 = GraphElement(node_id=node_id, label=label, weight=weight)
-        elem2 = GraphElement(node_id=node_id, label=label, weight=weight)
+        assert elem.type == node_type
         
-        assert elem1 == elem2
-        assert hash(elem1) == hash(elem2)
+    @given(node_id=valid_id_strategy, label=short_label, node_type=node_type_strategy)
+    def test_property_hash_consistency(self, node_id, label, node_type):
+        elem1 = Node(id=node_id, label=label, type=node_type)
+        # For Node, equality (and hash) is based on 'id'. Label and type can differ.
+        elem2 = Node(id=node_id, label=label + "_diff", type=random.choice(list(NodeType)))
```

Ruff (0.11.9)

Undefined name `random`

(F821)